core/domain/exp_domain.py

```diff
@@ -2130,24 +2130,26 @@ def extract_content_id_from_choices(html: str) -> str:
                 assert isinstance(value, str)
                 return extract_content_id_from_choices(value)
             elif new_type == 'SetOfTranslatableHtmlContentIds':
-                # Here 'migrate_rule_inputs_and_answers' method calls itself
-                # recursively and because of this MyPy assumes its type as
-                # recursive, like if this method returns List[str] then MyPy
-                # assumes its type as List[List[str]]. So, because of this,
-                # MyPy throws an error. Thus to avoid the error, we used
-                # ignore here.
+                # Here we use MyPy ignore because
```

Comment:
No -- Because as mentioned in the comment this method is recursive. So, if we cast its return type to List[str] then MyPy expects its return type as List[List[str]]. And if we cast to List[List[str]] then MyPy expects it's type as List[List[List[str]]. cycle never ends.

Reply:
Overloads added!!